ios/RNIapIos.m

```diff
@@ -463,12 +501,87 @@ -(NSDictionary*)getProductObject:(SKProduct *)product {
      introductoryPricePaymentMode, @"introductoryPricePaymentModeIOS",
      introductoryPriceNumberOfPeriods, @"introductoryPriceNumberOfPeriodsIOS",
      introductoryPriceSubscriptionPeriod, @"introductoryPriceSubscriptionPeriodIOS",
+     discounts, @"discounts",
      nil
- ];
+  ];
 
   return obj;
 }
 
+- (NSMutableArray *)getDiscountData:(NSArray *)discounts {
+  NSMutableArray *mappedDiscounts = [NSMutableArray arrayWithCapacity:[discounts count]];
+  NSString *localizedPrice;
+  NSString *paymendMode;
+  NSString *subscriptionPeriods;
+  NSString *discountType;
+
+  if (@available(iOS 11.2, *)) {
+    for(SKProductDiscount *discount in discounts) {
+      NSNumberFormatter *formatter = [[NSNumberFormatter alloc] init];
+      formatter.numberStyle = NSNumberFormatterCurrencyStyle;
+      formatter.locale = discount.priceLocale;
+      localizedPrice = [formatter stringFromNumber:discount.price];
+      
+      switch (discount.paymentMode) {
+        case SKProductDiscountPaymentModeFreeTrial:
+          paymendMode = @"FREETRIAL";
+          break;
+        case SKProductDiscountPaymentModePayAsYouGo:
+          paymendMode = @"PAYASYOUGO";
+          break;
+        case SKProductDiscountPaymentModePayUpFront:
+          paymendMode = @"PAYUPFRONT";
+          break;
+        default:
+          paymendMode = @"";
+          break;
+      }
+      
+      switch (discount.subscriptionPeriod.unit) {
+        case SKProductPeriodUnitDay:
+          subscriptionPeriods = @"DAY";
+          break;
+        case SKProductPeriodUnitWeek:
+          subscriptionPeriods = @"WEEK";
+          break;
+        case SKProductPeriodUnitMonth:
+          subscriptionPeriods = @"MONTH";
+          break;
+        case SKProductPeriodUnitYear:
+          subscriptionPeriods = @"YEAR";
+          break;
+        default:
+          subscriptionPeriods = @"";
+      }
+
+      switch (discount.type) {
+        case SKProductDiscountTypeIntroductory:
+          discountType = @"INTRODUCTORY";
+          break;
+        case SKProductDiscountTypeSubscription:
+          discountType = @"SUBSCRIPTION";
+          break;
+        default:
+          discountType = @"";
+          break;
+      }
+
+      [mappedDiscounts addObject:[NSDictionary dictionaryWithObjectsAndKeys:
+        discount.identifier, @"identifier",
+        discountType, @"type",
+        [@(discount.numberOfPeriods) stringValue], @"numberOfPeriods",
```

Comment:
Please check https://github.com/dooboolab/react-native-iap/pull/454

I think there can be the same issue

Reply:
Ok, will change it soon.